sky/backends/backend_utils.py

```diff
@@ -2661,36 +2873,39 @@ def get_backend_from_handle(
     return backend
 
 
-def get_task_demands_dict(task: 'task_lib.Task') -> Optional[Dict[str, float]]:
-    """Returns the accelerator dict of the task"""
-    # TODO: CPU and other memory resources are not supported yet.
-    accelerator_dict = None
+def get_task_demands_dict(task: 'task_lib.Task') -> Dict[str, float]:
+    """Returns the resources dict of the task"""
+    # TODO: CPU and other memory resources are not supported yet
+    # except for sky serve controller task.
+    resources_dict = {
+        # We set CPU resource for sky serve controller to a smaller value
+        # to support a larger number of services.
+        'CPU': (serve_lib.SERVICES_TASK_CPU_DEMAND if
```

Comment:
If we use the ray resource to indicate memory usage, the problem is that it will keep pending if the controller is full, since our ray cluster has no autoscaler enabled... Then we will notice it is full after a successful `sky.launch` on the controller. For the current implementation, we could detect it before the `sky.launch` is executed.

Reply:
Also, another problem is that pending ray jobs will take the same amount of memory too.